apps/asap-server/src/handlers/research-output/index-handler.ts

```diff
@@ -28,24 +28,39 @@ export const indexResearchOutputHandler = (
       SquidexWebhookResearchOutputPayload
     >,
   ): Promise<void> => {
-    const researchOutput = await researchOutputController.fetchById(
-      event.detail.payload.id,
-    );
+    logger.debug(`Event ${event['detail-type']}`);
 
-    await algoliaIndex.saveObject({
-      ...researchOutput,
-      objectID: researchOutput.id,
-    });
+    try {
+      const researchOutput = await researchOutputController.fetchById(
+        event.detail.payload.id,
+      );
 
-    logger.info(`Saved Research-Output with ID ${researchOutput.id}`);
+      logger.debug(`Fetched research-output ${researchOutput.id}`);
+
+      await algoliaIndex.saveObject({
+        ...researchOutput,
+        objectID: researchOutput.id,
+      });
+
+      logger.debug(`Saved research-output ${researchOutput.id}`);
+    } catch (e) {
+      logger.debug(JSON.stringify(e));
+
+      await algoliaIndex.deleteObject(event.detail.payload.id);
```

Comment:
what if algolia.saveObject throws? or what if it is an intermittent squidex networking error? we have to make sure its a 404 - but i would also double check to see what squidex returns when the resource is not present or is a draft